src/values.ts

```diff
@@ -15,8 +15,8 @@
  * @category Object
  */
 
-export function values<T>(
-  source: Record<PropertyKey, T> | ArrayLike<T>
+export function values<P extends PropertyKey, T>(
```

Comment:
There's no point in defining a generic if it isn't then used to tie the input and output. Our current typing isn't good because an index property can't be defined as a Record, and we are probably failing because of contravariance, and your fix simply swaps it to covariance so it "falls into place" (i might be mixing up contra- and co- variance here).